controllers/k8ssandra/stop_dc_test.go

```diff
@@ -0,0 +1,362 @@
+package k8ssandra
+
+import (
+	"context"
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	reaperapi "github.com/k8ssandra/k8ssandra-operator/apis/reaper/v1alpha1"
+	stargateapi "github.com/k8ssandra/k8ssandra-operator/apis/stargate/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+	"github.com/k8ssandra/k8ssandra-operator/test/framework"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+)
+
+// stopDc tests scenarios that involve stopping a CassandraDatacenter.
+func stopDc(t *testing.T, ctx context.Context, f *framework.Framework, _ string) {
+	t.Run("StopExistingDatacenter", stopDcTest(f, ctx, stopExistingDc))
+	t.Run("AddAndStopDatacenter", stopDcTest(f, ctx, addAndStopDc))
```

Comment:
I'm not sure this test is useful, I might remove it. There is nothing particular about stopping a dc that wasn't part of the initial cluster.